application/controllers/Os.php

```diff
@@ -358,14 +358,13 @@ public function gerarPagamentoGerencianetBoleto()
                 $this->session->set_flashdata('success', 'Cobrança criada com sucesso!');
             }
         } else {
-            if ($obj->code >= 40000) {
+            if($obj->code >= 40000){
```

Comment:
Acha melhor no local desse if fosse assim dentro do `else` na linha 360?
```

if ($obj->code != 200) {
                $json = ['code' => $obj->code, 'error' => $obj->error, 'errorDescription' => $obj->errorDescription];
                print_r(json_encode($json));
                return;
            } else {
                $json = ['code' => 4001, 'error' => 'server_error', 'errorDescription' => 'Falha ao gerar cobrança/link verifique a conexão com a internet'];
            return print_r(json_encode($json));
            }
```

Reply:
se for diferente de 200 ele ja mostra o codigo do erro e a mensagem do erro?

Reply:
vdd da de simplificar né, vou da uma olhada